components/app_bar/app_bar.tsx

```diff
@@ -23,9 +24,40 @@ export default function AppBar() {
         return null;
     }
 
+    let playbooksComponent;
+    let boardsComponent;
+
+    // Extract the Playbooks and Boards components so they can be reordered at the beginning of the list
+    const filteredComponents = appBarPluginComponents.filter((component) => {
```

Comment:
Could we just use a sort instead of pulling these out with a filter and re-adding them? Having a filter with side effects also feels not good.

Sorting is stable, so as long as we sort playbooks and boards before everything else, then everything else should remain in the same order.

Reply:
> Having a filter with side effects also feels not good

I agree, I first did a filter to keep all but playbooks and boards and then iterated again to retrieve playbooks and boards, but then decided to do everything in one loop.

With sorting, you mean using `sort` and checking the plugin IDs there? It can work, too, maybe it's more elegant.

Reply:
+1 for sorting instead of filtering

Reply:
Done in https://github.com/mattermost/mattermost-webapp/pull/10311/commits/3e0404f0a8b15192589f8d444ee70dc8f96183de. Much cleaner, thanks!